cmd/mdatagen/internal/templates/telemetry.go.tmpl

```diff
@@ -76,6 +84,10 @@ func (builder *TelemetryBuilder) Register{{ $name.Render }}Callback(cb metric.{{
     defer builder.mu.Unlock()
     builder.registrations = append(builder.registrations, reg)
     return nil
+}
+    {{- else }}
+func (builder *TelemetryBuilder) Record{{ $name.Render }}DataPoint(ctx context.Context, val {{ $metric.Data.BasicType }}) {
```

Comment:
We probably shouldn't hide `Add` behind `Record`. See a previous effort https://github.com/open-telemetry/opentelemetry-collector/pull/12182#discussion_r1929576706